server/src/security/createUserValidationFunc.js

```diff
@@ -47,22 +47,31 @@ const createUserValidationFunc = (db, audience, auth0ManagementClient) => {
                         debug('User with info', userInfo)
                     } catch (e) {
                         debug(e)
-                        throw Boom.conflict(
+                        throw Boom.unauthorized(
                             'Failed to get user information from Identity Provider'
                         )
                     }
                     if (!userInfo.email_verified) {
-                        throw Boom.conflict('Email not verified')
+                        throw Boom.unauthorized('Email not verified')
                     }
                     //create the user if it doesn't exist
                     const createUserTransaction = async trx => {
-                        const dbUser = await createUser(
-                            {
-                                email: userInfo.email,
-                                name: userInfo.name,
-                            },
-                            trx
-                        )
+                        let dbUser
+                        try {
+                            dbUser = await createUser(
+                                {
+                                    email: userInfo.email,
+                                    name: userInfo.name,
+                                },
+                                trx
+                            )
+                        } catch (e) {
+                            // if user exist and not current external-id throw a specific error
+                            Bounce.ignore(wrapError(e), UniqueViolationError)
+                            throw Boom.unauthorized(
+                                'Email is already registered with another provider.'
```

Comment:
Should maybe add "please contact DHIS2 at <xyz> if you think this is a mistake."